src/pages/AssignServerRoles.js

```diff
@@ -1266,17 +1304,28 @@ class AssignServerRoles extends BaseWizardPage {
       }
     });
     let dupIds = [...new Set(tempDups)];
+    let theProps = {};
+    // for install
+    if(this.props.mode === undefined) {
+      theProps.editAction = this.handleShowEditServer;
+      theProps.deleteAction = this.handleDeleteServer;
+    }
+    else if (this.props.mode === 'addserver') {
+      // currently don't have editAction and deleteAction
+      // once we figure out how to differentiate deployed servers from
+      // newly added servers then we will add the actions to the newly added
+      // servers
+      theProps.mode = this.props.mode;
+    }
     return (
       <ServerRolesAccordion
         ondropFunct={this.assignServerToRoleDnD}
         ondragEnterFunct={this.highlightDrop}
         ondragLeaveFunct={this.unHighlightDrop}
         allowDropFunct={this.allowDrop}
-        serverRoles={getServerRoles(this.props.model)}
-        tableId='rightTableId' checkDupIds={dupIds}
-        editAction={this.handleShowEditServer}
-        viewAction={this.handleShowServerDetails}
-        deleteAction={this.handleDeleteServer}>
+        serverRoles={getServerRoles(this.props.model, this.props.rolesLimit)}
+        tableId='rightTableId' checkDupIds={dupIds} {...theProps}
```

Comment:
I'd probably move {...theProps} to its own line so its more obvious that its a standalone set of key/value pairs... maybe rename it as well... extraProps or something... "theProps" had me confused whether it was duplicating the current objects props or something passed down to the next object

Reply:
ok will update